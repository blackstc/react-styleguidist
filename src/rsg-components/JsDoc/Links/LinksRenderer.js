import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';
import Markdown from 'rsg-components/Markdown';

export const styles = ({ font }) => ({
	linksBlock: {
		marginTop: 6,
		fontSize: 'inherit',
	},
	linksHeader: {
		fontSize: 'inherit',
		fontFamily: font,
	},
	linksList: {
		fontSize: 'inherit',
		marginTop: 0,
		marginBottom: 0,
		marginLeft: 20,
	},
});

export function LinksRenderer({ classes, tags }) {
	if (!tags || (!tags.link && !tags.see)) {
		return null;
	}

	function renderLinks(links) {
		return links.map((link, index) => {
			return (
				<li key={`link-${index}`}><Markdown text={link.description} inline /></li>
			);
		});
	}

	return (
		<div className={classes.linksBlock}>
			<div className={classes.linksHeader}>More information</div>
			<ul className={classes.linksList}>
				{ tags.see && renderLinks(tags.see) }
				{ tags.link && renderLinks(tags.link) }
			</ul>
		</div>

	);
}
LinksRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	tags: PropTypes.shape({
		see: PropTypes.array,
		link: PropTypes.array,
	}).isRequired,
};
export default Styled(styles)(LinksRenderer);
