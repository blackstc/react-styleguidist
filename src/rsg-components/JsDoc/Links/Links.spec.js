import React from 'react';
import { LinksRenderer } from './LinksRenderer';

describe('Links', () => {
	function createTestTag(title, description = 'Test description') {
		return {
			title,
			description,
		};
	}

	const linkTags = {
		link: [createTestTag('link'), createTestTag('link', '[Test](#TestLink)')],
	};
	const seeTags = {
		see: [createTestTag('see'), createTestTag('see', 'Test description 2')],
	};

	it('renderer should render for one @link tag', () => {
		const tags = {
			link: [linkTags.link[0]],
		};

		const actual = render(
			<LinksRenderer classes={{}} tags={tags} />
		);

		expect(actual).toMatchSnapshot();
	});

	it('renderer should render for two @link tags', () => {
		const actual = render(
			<LinksRenderer classes={{}} tags={linkTags} />
		);

		expect(actual).toMatchSnapshot();
	});

	it('renderer should render for two @see tags', () => {
		const actual = render(
			<LinksRenderer classes={{}} tags={seeTags} />
		);

		expect(actual).toMatchSnapshot();
	});

	it('renderer should render for mixed @see and @link tags', () => {
		const tags = {
			...linkTags,
			...seeTags,
		};

		const actual = render(
			<LinksRenderer classes={{}} tags={tags} />
		);

		expect(actual).toMatchSnapshot();
	});


	it('renderer should not render for empty tags', () => {
		const actual = shallow(
			<LinksRenderer classes={{}} tags={{}} />
		);

		expect(actual.node).toBe(null);
	});
});

