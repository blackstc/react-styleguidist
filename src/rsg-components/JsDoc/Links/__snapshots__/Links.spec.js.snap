// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Links renderer should render for mixed @see and @link tags 1`] = `
<div>
  <div>
    More information
  </div>
  <ul>
    <li>
      <span
        class="base-0-0"
      >
        Test description
      </span>
    </li>
    <li>
      <span
        class="base-0-0"
      >
        Test description 2
      </span>
    </li>
    <li>
      <span
        class="base-0-0"
      >
        Test description
      </span>
    </li>
    <li>
      <span
        class="base-0-0"
      >
        <a
          class="a-0-2"
          href="#TestLink"
        >
          Test
        </a>
      </span>
    </li>
  </ul>
</div>
`;

exports[`Links renderer should render for one @link tag 1`] = `
<div>
  <div>
    More information
  </div>
  <ul>
    <li>
      <span
        class="base-0-0"
      >
        Test description
      </span>
    </li>
  </ul>
</div>
`;

exports[`Links renderer should render for two @link tags 1`] = `
<div>
  <div>
    More information
  </div>
  <ul>
    <li>
      <span
        class="base-0-0"
      >
        Test description
      </span>
    </li>
    <li>
      <span
        class="base-0-0"
      >
        <a
          class="a-0-2"
          href="#TestLink"
        >
          Test
        </a>
      </span>
    </li>
  </ul>
</div>
`;

exports[`Links renderer should render for two @see tags 1`] = `
<div>
  <div>
    More information
  </div>
  <ul>
    <li>
      <span
        class="base-0-0"
      >
        Test description
      </span>
    </li>
    <li>
      <span
        class="base-0-0"
      >
        Test description 2
      </span>
    </li>
  </ul>
</div>
`;
